pipeline-steps/templates/jobs/fortify-scan-job.yaml

```diff
@@ -17,22 +19,43 @@ parameters:
 jobs:
   - ${{ if or(eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(variables['Build.SourceBranch'], 'refs/heads/main'), startsWith(variables['Build.SourceBranch'], 'refs/pull/')) }}:
     - job: PrepareFortifySecrets
+      variables:
+        - group: ${{ parameters.variableGroup }}
```

Comment:
Due to the variable group being processed at runtime compared to the expressions at compile, it can't determine the existance of the fortify secret variables prior to assigning the pool.
So instead we have the useVariableGroup parameter which flags to use variable group or not, defaulting to true. If true it'll use an Azure agent ubuntu-latest to retrieve the variables stored in mi-env-variables.